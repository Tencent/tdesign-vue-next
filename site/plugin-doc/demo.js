/* eslint-disable */
import path from 'path';

export default function renderDemo(md, container) {
  md.use(container, 'demo', {
    validate(params) {
      return params.trim().match(/^demo\s+([\\/.\w-]+)(\s+(.+?))?(\s+--dev)?$/);
    },
    render(tokens, idx) {
      if (tokens[idx].nesting === 1) {
        const match = tokens[idx].info.trim().match(/^demo\s+([\\/.\w-]+)(\s+(.+?))?(\s+--dev)?$/);
        const [, demoPath, componentName = ''] = match;
        const demoPathOnlyLetters = demoPath.replace(/[^a-zA-Z\d]/g, '');
        const demoName = path.basename(demoPath);
        const demoDefName = `Demo${demoPathOnlyLetters}`;
        const demoCodeDefName = `Demo${demoPathOnlyLetters}Code`;
        const tpl = `
          <td-doc-demo component-name="${componentName.trim()}" :code=${demoCodeDefName} demo-name="${demoName}" language="markup">
            <div slot="action">
              <Stackblitz demo-name="${demoName}" component-name="${componentName}" :code=${demoCodeDefName} />
              <CodeSandbox demo-name="${demoName}" component-name="${componentName}" :code=${demoCodeDefName} />
            </div>
            <div class="tdesign-demo-item__body">
              <${demoDefName} />
            </div>
          </td-doc-demo>
        `;

        tokens.tttpl = tpl;

        return `<div class="tdesign-demo-wrapper tdesign-demo-item--${componentName.trim()}-${demoName} tdesign-demo-item--${componentName.trim()}">`;
      }
      if (tokens.tttpl) {
        return `${tokens.tttpl || ''}</div>`;
      }
      return '';
    },
  });
}
