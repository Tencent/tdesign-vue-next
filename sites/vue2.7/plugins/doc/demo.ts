import * as path from 'path';

export default function renderDemo(md, container) {
  md.use(container, 'demo', {
    validate(params) {
      return params.trim().match(/^demo\s+([\\/.\w-]+)(\s+(.+?))?(\s+--dev)?$/);
    },
    render(tokens, idx) {
      if (tokens[idx].nesting === 1) {
        const match = tokens[idx].info.trim().match(/^demo\s+([\\/.\w-]+)(\s+(.+?))?(\s+--dev)?$/);
        const [, demoPath, componentName = ''] = match;
        const demoPathOnlyLetters = demoPath.replace(/[^a-zA-Z\d]/g, '');
        const demoName = path.basename(demoPath);
        const demoDefName = `Demo${demoPathOnlyLetters}`;
        const demoCodeDefName = `Demo${demoPathOnlyLetters}Code`;
        const demoCompositionCodeDefName = `Demo${demoPathOnlyLetters}CompositionCode`; // composition示例

        const tpl = `
          <td-doc-demo demo-name="${demoName}" languages="OptionalAPI,CompositionAPI" :data-OptionalAPI="${demoCodeDefName}" :data-CompositionAPI="${demoCompositionCodeDefName}" language="markup">            <div slot="action">
              <Stackblitz demo-name="${demoName}" component-name="${componentName}" />
              <CodeSandbox demo-name="${demoName}" component-name="${componentName}" />
            </div>
            <div class="tdesign-demo-item__body">
              <${demoDefName} />
            </div>
          </td-doc-demo>
        `;

        tokens.tttpl = tpl;

        return `<div class="tdesign-demo-wrapper tdesign-demo-item--${componentName.trim()}-${demoName} tdesign-demo-item--${componentName.trim()}">`;
      }
      if (tokens.tttpl) {
        return `${tokens.tttpl || ''}</div>`;
      }
      return '';
    },
  });
}
