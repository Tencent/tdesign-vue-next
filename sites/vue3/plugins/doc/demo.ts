import * as path from "path";

export default function renderDemo(md, container) {
  md.use(container, "demo", {
    validate(params) {
      return params.trim().match(/^demo\s+([\\/.\w-]+)(\s+(.+?))?(\s+--dev)?$/);
    },
    render(tokens, idx) {
      if (tokens[idx].nesting === 1) {
        const match = tokens[idx].info
          .trim()
          .match(/^demo\s+([\\/.\w-]+)(\s+(.+?))?(\s+--dev)?$/);
        const [, demoPath, componentName = ""] = match;
        const demoPathOnlyLetters = demoPath.replace(/[^a-zA-Z\d]/g, "");
        const demoName = path.basename(demoPath);
        const demoDefName = `Demo${demoPathOnlyLetters}`;
        const demoCodeDefName = `Demo${demoPathOnlyLetters}Code`;
        const demoTSCodeDefName = `Demo${demoPathOnlyLetters}TsCode`; // ts示例
        const tpl = `
          <td-doc-demo component-name="${componentName.trim()}" demo-name="${demoName}" language="markup" languages="JavaScript,TypeScript" :data-JavaScript="${demoCodeDefName}" :data-TypeScript="${demoTSCodeDefName}">
            <div slot="action">
              <Stackblitz demo-name="${demoName}" component-name="${componentName}" />
              <CodeSandbox demo-name="${demoName}" component-name="${componentName}" />
            </div>
            <div class="tdesign-demo-item__body">
              <${demoDefName} />
            </div>
          </td-doc-demo>
        `;

        tokens.tttpl = tpl;

        return `<div class="tdesign-demo-wrapper tdesign-demo-item--${componentName.trim()}-${demoName} tdesign-demo-item--${componentName.trim()}">`;
      }
      if (tokens.tttpl) {
        return `${tokens.tttpl || ""}</div>`;
      }
      return "";
    },
  });
}
