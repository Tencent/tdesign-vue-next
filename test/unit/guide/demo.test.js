import { mount } from '@vue/test-utils';
import demo from '@/examples/guide/demos/base.vue';

// unit test for component in examples.
describe('Guide', () => {
  it('base demo works fine', () => {
    const wrapper = mount(demo);
    expect(wrapper.element).toMatchSnapshot();
  });
});
