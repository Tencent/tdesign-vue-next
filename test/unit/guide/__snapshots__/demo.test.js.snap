// Vitest Snapshot v1

exports[`Guide > base demo works fine 1`] = `
<t-row
  justify="center"
>
  <t-button>
    新手引导
  </t-button>
  <t-drawer
    destroy-on-close=""
    header="演示新手引导"
    show-overlay="false"
    size="60%"
    visible="false"
  >
    <div
      class="guide-container"
    >
      <div
        class="main-title"
      >
        <div
          class="title-major"
        >
          Guide 用户引导
        </div>
        <div
          class="title-sub"
        >
          按钮用于开启一个闭环的操作任务，如“删除”对象、“购买”商品等。
        </div>
      </div>
      <div
        class="field label-field"
      >
        <div
          class="label"
        >
          Label
        </div>
        <t-input
          placeholder="请输入内容"
        />
      </div>
      <div
        class="field"
      >
        <div
          class="label"
        >
          Label
        </div>
        <t-input
          placeholder="请输入内容"
        />
      </div>
      <t-row
        class="action"
      >
        <t-button>
          确定
        </t-button>
        <t-button
          theme="default"
          variant="base"
        >
          取消
        </t-button>
      </t-row>
    </div>
    <t-guide
      modelvalue="-1"
      steps="[object Object],[object Object],[object Object]"
    />
  </t-drawer>
</t-row>
`;
